src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java

```diff
@@ -207,14 +199,11 @@ private void applyMappingsForAddress(
     }
 
     private void addMappingsToChildren(List<OcrDataField> ocrDataFields, Map<String, Object> modifiedMap) {
-        List<ImmutableMap<String, Object>> children = new ArrayList<>();
-        for (int i = 1; ; i++) {
-            if (isChildInfoPopulated(i, ocrDataFields)) {
-                children.add(mapChild(i, ocrDataFields));
-            } else {
-                // there is no child with given index, stop mapping
-                break;
-            }
+        ChildrenList children = new ChildrenList();
+        int i = 1;
+        while (isChildInfoPopulated(i, ocrDataFields)) {
```

Comment:
If it's not possible to solve this with streams, I'd personally prefer the "for" version (but without the "break", just stopping at the end of the list).
But I'm confident this can be solved with streams.

Reply:
there is no "end of the list" :)

We have a given number of children in OCR field list, but is like:

```
{
  "case_type_id": "FinancialRemedyContested",
  "id": "LV481297",
  "po_box": "PO 17",
  "form_type": "formA",
  "ocr_data_fields": [
    ...
    { "name": "AuthorisationSolicitorPosition", "value": "I'm the CEO" },
    { "name": "NameOfChild1", "value": "Bob" },
    { "name": "RelationshipToApplicantChild1", "value": "Father" },
    { "name": "RelationshipToRespondentChild1", "value": "Mother" },
    { "name": "CountryOfResidenceChild1", "value": "UK" },
    { "name": "DateOfBirthChild1", "value": "03/06/2011" },
    { "name": "GenderChild1", "value": "male" },
    { "name": "NameOfChild2", "value": "Tommy" },
    { "name": "RelationshipToApplicantChild2", "value": "Mother" },
    { "name": "RelationshipToRespondentChild2", "value": "" },
    { "name": "CountryOfResidenceChild2", "value": "FR" },
    { "name": "DateOfBirthChild2", "value": "04/06/2010" },
    { "name": "GenderChild2", "value": "female" }
  // here we could have XXXchild3, XXXChild4, etc, but we never know as long as we check this out 
  // I have a method that checks if there is `"NameOfChild" + i` populated -> it's my assumption, if there is no name - there is no child number i :)
  ]
```

Reply:
Got you